third_party/gtp_ovs/ovs-gtp-patches/2.15/0018-GTP-seq-number-length-correction.patch

```diff
@@ -0,0 +1,36 @@
+From b30568295e28a8918a7def614861909b29219c8b Mon Sep 17 00:00:00 2001
+From: Yogesh Pandey <yogesh@wavelabs.ai>
+Date: Sat, 21 Aug 2021 09:58:46 +0000
+Subject: [PATCH] gtp: Fix for sequence number length calculation
+
+Issue:
+  Additional 4 bytes getting added in header length
+  if seq_flag is set to true in gtp header
+
+Fix:
+  Ir-respective if the flag is set or not the length
+  of gtpu header is fixed which includes seq no.
+
+Signed-off-by: Yogesh Pandey <yogesh@wavelabs.ai>
+---
+ datapath/linux/compat/gtp.c | 4 ----
+ 1 file changed, 4 deletions(-)
+
+diff --git a/datapath/linux/compat/gtp.c b/datapath/linux/compat/gtp.c
+index 5046bd6..bc2569f 100644
+--- a/datapath/linux/compat/gtp.c
++++ b/datapath/linux/compat/gtp.c
+@@ -101,10 +101,6 @@ static int gtp_rx(struct gtp_dev *gtp, struct sk_buff *skb,
+         int opts_len = 0;
+         if (unlikely(type != GTP_TPDU)) {
+             opts_len = sizeof (struct gtpu_metadata);
+-        } else {
```

Comment:
This might break GTP packets with sequence headers but without extension header. lets move the header adjustment as done here: https://github.com/magma/magma/blob/19f41f677717cd1066c35cac9aaad66c7ec815a3/third_party/gtp_ovs/kernel-4.9/gtp-v4.9-backport/gtp.c#L361

Reply:
Comments addressed. Tested three scenarios :
  1. 
  [  938.306345] gtpu_sys_2152: flags 36 type: ff
  [  938.306403] gtpu_sys_2152: attaching metadata_dst to skb, gtp ver 1 hdrlen 24
  [  938.306422] gtpu_sys_2152: inner pkt: ipv4

2.
[  938.357149] gtpu_sys_2152: flags 30 type: ff
[  938.357160] gtpu_sys_2152: attaching metadata_dst to skb, gtp ver 1 hdrlen 16
[  938.357177] gtpu_sys_2152: inner pkt: ipv4

3.
[  938.413639] gtpu_sys_2152: flags 34 type: ff
[  938.413677] gtpu_sys_2152: attaching metadata_dst to skb, gtp ver 1 hdrlen 24
[  938.413694] gtpu_sys_2152: inner pkt: ipv4